pkg/controllers/cspc-controller/util/cspc_util.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NewCSPCCondition creates a new cspc condition.
+func NewCSPCCondition(condType cstor.CSPCConditionType, status corev1.ConditionStatus, reason, message string) *cstor.CStorPoolClusterCondition {
```

Comment:
Yes.. Pointer is fine.

Reply:
Poiner is returned so that the change that we do to the object is persisted. 
We pass this object to other functions that can do the change -- not returning pointer will mean unnnecessary dereferencing.